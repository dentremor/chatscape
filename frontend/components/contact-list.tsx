import {
  Avatar,
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateVariant,
  List,
  ListItem,
  Skeleton,
} from "@patternfly/react-core";

export const ContactList = ({
  contacts,
  contactList,
  width,
  activeContactID,
  setActiveContactID,
  setDrawerExpanded,
}: {
  contacts: {
    id: string;
    avatar: string;
    name: string;
    intro: string;
  }[];
  contactList: {
    id: string;
    avatar: string;
    name: string;
    intro: string;
  }[];
  width?: number;
  activeContactID: string | null;
  setActiveContactID: (id: string) => void;
  setDrawerExpanded: (expanded: boolean) => void;
}) => (
  <List isPlain>
    {contacts.length > 0 ? (
      contactList && contactList.length > 0 ? (
        contactList.map((contact, i) => (
          <ListItem key={i}>
            <Button
              variant="plain"
              className="pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
              isActive={contact.id === activeContactID}
              onClick={() => {
                setActiveContactID(contact.id);

                if (!(!width || width >= 768)) {
                  setDrawerExpanded(false);
                }
              }}
            >
              <Avatar
                src={contact.avatar}
                alt="avatar"
                className="pf-u-mr-md"
              />

              <div className="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center">
                <div className="pf-u-font-family-heading-sans-serif">
                  {contact.name}
                </div>

                <div className="pf-u-icon-color-light">{contact.intro} ...</div>
              </div>
            </Button>
          </ListItem>
        ))
      ) : (
        <ListItem>
          <EmptyState variant={EmptyStateVariant.xs}>
            <EmptyStateBody>No contacts found</EmptyStateBody>
          </EmptyState>
        </ListItem>
      )
    ) : (
      [0, 1, 2].map((_, i) => (
        <ListItem key={i}>
          <Button
            variant="plain"
            className="pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
            disabled
          >
            <Skeleton
              shape="circle"
              width="36px"
              height="36px"
              screenreaderText="Loading avatar"
              className="pf-u-mr-md"
            />
            <div className="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center pf-u-flex-1">
              <div className="pf-u-font-family-heading-sans-serif pf-u-w-100 pf-u-pb-sm">
                <Skeleton
                  screenreaderText="Loading contact info"
                  width={(i % 2 !== 0 ? 33 : 66) + (i % 2) * 10 + "%"}
                />
              </div>
              <div className="pf-u-icon-color-light pf-u-w-100">
                <Skeleton
                  screenreaderText="Loading contact info"
                  width={(i % 2 === 0 ? 33 : 66) + (i % 2) * 10 + "%"}
                />
              </div>
            </div>
          </Button>
        </ListItem>
      ))
    )}
  </List>
);
