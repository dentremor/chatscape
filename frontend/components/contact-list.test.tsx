import { render } from "@testing-library/react";
import { ContactList } from "./contact-list";

const mockContacts = [
  { id: "1", avatar: "avatar1.png", name: "Contact 1", intro: "Intro 1" },
  { id: "2", avatar: "avatar2.png", name: "Contact 2", intro: "Intro 2" },
];

describe("ContactList", () => {
  it("renders with empty contacts list", () => {
    expect(
      render(
        <ContactList
          contacts={[]}
          contactList={[]}
          activeContactID={null}
          setActiveContactID={jest.fn()}
          setDrawerExpanded={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders with full contacts list", () => {
    expect(
      render(
        <ContactList
          contacts={mockContacts}
          contactList={[]}
          activeContactID={null}
          setActiveContactID={jest.fn()}
          setDrawerExpanded={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders with full contacts and contactList", () => {
    expect(
      render(
        <ContactList
          contacts={mockContacts}
          contactList={mockContacts}
          activeContactID={null}
          setActiveContactID={jest.fn()}
          setDrawerExpanded={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders with a set activeContactID", () => {
    expect(
      render(
        <ContactList
          contacts={mockContacts}
          contactList={mockContacts}
          activeContactID={mockContacts[0].id}
          setActiveContactID={jest.fn()}
          setDrawerExpanded={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });
});
