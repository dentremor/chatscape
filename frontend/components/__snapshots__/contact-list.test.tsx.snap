// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList renders with a set activeContactID 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="pf-c-list pf-m-plain"
      >
        <li
          class=""
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain pf-m-active pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
            data-ouia-component-id="OUIA-Generated-Button-plain-6"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <img
              alt="avatar"
              class="pf-c-avatar pf-u-mr-md"
              src="avatar1.png"
            />
            <div
              class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center"
            >
              <div
                class="pf-u-font-family-heading-sans-serif"
              >
                Contact 1
              </div>
              <div
                class="pf-u-icon-color-light"
              >
                Intro 1
                 ...
              </div>
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
            data-ouia-component-id="OUIA-Generated-Button-plain-7"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <img
              alt="avatar"
              class="pf-c-avatar pf-u-mr-md"
              src="avatar2.png"
            />
            <div
              class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center"
            >
              <div
                class="pf-u-font-family-heading-sans-serif"
              >
                Contact 2
              </div>
              <div
                class="pf-u-icon-color-light"
              >
                Intro 2
                 ...
              </div>
            </div>
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="pf-c-list pf-m-plain"
    >
      <li
        class=""
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain pf-m-active pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
          data-ouia-component-id="OUIA-Generated-Button-plain-6"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <img
            alt="avatar"
            class="pf-c-avatar pf-u-mr-md"
            src="avatar1.png"
          />
          <div
            class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center"
          >
            <div
              class="pf-u-font-family-heading-sans-serif"
            >
              Contact 1
            </div>
            <div
              class="pf-u-icon-color-light"
            >
              Intro 1
               ...
            </div>
          </div>
        </button>
      </li>
      <li
        class=""
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
          data-ouia-component-id="OUIA-Generated-Button-plain-7"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <img
            alt="avatar"
            class="pf-c-avatar pf-u-mr-md"
            src="avatar2.png"
          />
          <div
            class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center"
          >
            <div
              class="pf-u-font-family-heading-sans-serif"
            >
              Contact 2
            </div>
            <div
              class="pf-u-icon-color-light"
            >
              Intro 2
               ...
            </div>
          </div>
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ContactList renders with empty contacts list 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="pf-c-list pf-m-plain"
      >
        <li
          class=""
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <div
              class="pf-c-skeleton pf-m-circle pf-u-mr-md"
              style="--pf-c-skeleton--Width: 36px; --pf-c-skeleton--Height: 36px;"
            >
              <span
                class="pf-u-screen-reader"
              >
                Loading avatar
              </span>
            </div>
            <div
              class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center pf-u-flex-1"
            >
              <div
                class="pf-u-font-family-heading-sans-serif pf-u-w-100 pf-u-pb-sm"
              >
                <div
                  class="pf-c-skeleton"
                  style="--pf-c-skeleton--Width: 66%;"
                >
                  <span
                    class="pf-u-screen-reader"
                  >
                    Loading contact info
                  </span>
                </div>
              </div>
              <div
                class="pf-u-icon-color-light pf-u-w-100"
              >
                <div
                  class="pf-c-skeleton"
                  style="--pf-c-skeleton--Width: 33%;"
                >
                  <span
                    class="pf-u-screen-reader"
                  >
                    Loading contact info
                  </span>
                </div>
              </div>
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
            data-ouia-component-id="OUIA-Generated-Button-plain-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <div
              class="pf-c-skeleton pf-m-circle pf-u-mr-md"
              style="--pf-c-skeleton--Width: 36px; --pf-c-skeleton--Height: 36px;"
            >
              <span
                class="pf-u-screen-reader"
              >
                Loading avatar
              </span>
            </div>
            <div
              class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center pf-u-flex-1"
            >
              <div
                class="pf-u-font-family-heading-sans-serif pf-u-w-100 pf-u-pb-sm"
              >
                <div
                  class="pf-c-skeleton"
                  style="--pf-c-skeleton--Width: 43%;"
                >
                  <span
                    class="pf-u-screen-reader"
                  >
                    Loading contact info
                  </span>
                </div>
              </div>
              <div
                class="pf-u-icon-color-light pf-u-w-100"
              >
                <div
                  class="pf-c-skeleton"
                  style="--pf-c-skeleton--Width: 76%;"
                >
                  <span
                    class="pf-u-screen-reader"
                  >
                    Loading contact info
                  </span>
                </div>
              </div>
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
            data-ouia-component-id="OUIA-Generated-Button-plain-3"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <div
              class="pf-c-skeleton pf-m-circle pf-u-mr-md"
              style="--pf-c-skeleton--Width: 36px; --pf-c-skeleton--Height: 36px;"
            >
              <span
                class="pf-u-screen-reader"
              >
                Loading avatar
              </span>
            </div>
            <div
              class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center pf-u-flex-1"
            >
              <div
                class="pf-u-font-family-heading-sans-serif pf-u-w-100 pf-u-pb-sm"
              >
                <div
                  class="pf-c-skeleton"
                  style="--pf-c-skeleton--Width: 66%;"
                >
                  <span
                    class="pf-u-screen-reader"
                  >
                    Loading contact info
                  </span>
                </div>
              </div>
              <div
                class="pf-u-icon-color-light pf-u-w-100"
              >
                <div
                  class="pf-c-skeleton"
                  style="--pf-c-skeleton--Width: 33%;"
                >
                  <span
                    class="pf-u-screen-reader"
                  >
                    Loading contact info
                  </span>
                </div>
              </div>
            </div>
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="pf-c-list pf-m-plain"
    >
      <li
        class=""
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
          data-ouia-component-id="OUIA-Generated-Button-plain-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <div
            class="pf-c-skeleton pf-m-circle pf-u-mr-md"
            style="--pf-c-skeleton--Width: 36px; --pf-c-skeleton--Height: 36px;"
          >
            <span
              class="pf-u-screen-reader"
            >
              Loading avatar
            </span>
          </div>
          <div
            class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center pf-u-flex-1"
          >
            <div
              class="pf-u-font-family-heading-sans-serif pf-u-w-100 pf-u-pb-sm"
            >
              <div
                class="pf-c-skeleton"
                style="--pf-c-skeleton--Width: 66%;"
              >
                <span
                  class="pf-u-screen-reader"
                >
                  Loading contact info
                </span>
              </div>
            </div>
            <div
              class="pf-u-icon-color-light pf-u-w-100"
            >
              <div
                class="pf-c-skeleton"
                style="--pf-c-skeleton--Width: 33%;"
              >
                <span
                  class="pf-u-screen-reader"
                >
                  Loading contact info
                </span>
              </div>
            </div>
          </div>
        </button>
      </li>
      <li
        class=""
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
          data-ouia-component-id="OUIA-Generated-Button-plain-2"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <div
            class="pf-c-skeleton pf-m-circle pf-u-mr-md"
            style="--pf-c-skeleton--Width: 36px; --pf-c-skeleton--Height: 36px;"
          >
            <span
              class="pf-u-screen-reader"
            >
              Loading avatar
            </span>
          </div>
          <div
            class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center pf-u-flex-1"
          >
            <div
              class="pf-u-font-family-heading-sans-serif pf-u-w-100 pf-u-pb-sm"
            >
              <div
                class="pf-c-skeleton"
                style="--pf-c-skeleton--Width: 43%;"
              >
                <span
                  class="pf-u-screen-reader"
                >
                  Loading contact info
                </span>
              </div>
            </div>
            <div
              class="pf-u-icon-color-light pf-u-w-100"
            >
              <div
                class="pf-c-skeleton"
                style="--pf-c-skeleton--Width: 76%;"
              >
                <span
                  class="pf-u-screen-reader"
                >
                  Loading contact info
                </span>
              </div>
            </div>
          </div>
        </button>
      </li>
      <li
        class=""
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
          data-ouia-component-id="OUIA-Generated-Button-plain-3"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <div
            class="pf-c-skeleton pf-m-circle pf-u-mr-md"
            style="--pf-c-skeleton--Width: 36px; --pf-c-skeleton--Height: 36px;"
          >
            <span
              class="pf-u-screen-reader"
            >
              Loading avatar
            </span>
          </div>
          <div
            class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center pf-u-flex-1"
          >
            <div
              class="pf-u-font-family-heading-sans-serif pf-u-w-100 pf-u-pb-sm"
            >
              <div
                class="pf-c-skeleton"
                style="--pf-c-skeleton--Width: 66%;"
              >
                <span
                  class="pf-u-screen-reader"
                >
                  Loading contact info
                </span>
              </div>
            </div>
            <div
              class="pf-u-icon-color-light pf-u-w-100"
            >
              <div
                class="pf-c-skeleton"
                style="--pf-c-skeleton--Width: 33%;"
              >
                <span
                  class="pf-u-screen-reader"
                >
                  Loading contact info
                </span>
              </div>
            </div>
          </div>
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ContactList renders with full contacts and contactList 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="pf-c-list pf-m-plain"
      >
        <li
          class=""
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
            data-ouia-component-id="OUIA-Generated-Button-plain-4"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <img
              alt="avatar"
              class="pf-c-avatar pf-u-mr-md"
              src="avatar1.png"
            />
            <div
              class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center"
            >
              <div
                class="pf-u-font-family-heading-sans-serif"
              >
                Contact 1
              </div>
              <div
                class="pf-u-icon-color-light"
              >
                Intro 1
                 ...
              </div>
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
            data-ouia-component-id="OUIA-Generated-Button-plain-5"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <img
              alt="avatar"
              class="pf-c-avatar pf-u-mr-md"
              src="avatar2.png"
            />
            <div
              class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center"
            >
              <div
                class="pf-u-font-family-heading-sans-serif"
              >
                Contact 2
              </div>
              <div
                class="pf-u-icon-color-light"
              >
                Intro 2
                 ...
              </div>
            </div>
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="pf-c-list pf-m-plain"
    >
      <li
        class=""
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
          data-ouia-component-id="OUIA-Generated-Button-plain-4"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <img
            alt="avatar"
            class="pf-c-avatar pf-u-mr-md"
            src="avatar1.png"
          />
          <div
            class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center"
          >
            <div
              class="pf-u-font-family-heading-sans-serif"
            >
              Contact 1
            </div>
            <div
              class="pf-u-icon-color-light"
            >
              Intro 1
               ...
            </div>
          </div>
        </button>
      </li>
      <li
        class=""
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain pf-u-display-flex pf-u-align-items-center pf-u-p-md pf-u-contact-selector pf-u-w-100"
          data-ouia-component-id="OUIA-Generated-Button-plain-5"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <img
            alt="avatar"
            class="pf-c-avatar pf-u-mr-md"
            src="avatar2.png"
          />
          <div
            class="pf-u-display-flex pf-u-flex-direction-column pf-u-align-items-flex-start pf-u-justify-content-center"
          >
            <div
              class="pf-u-font-family-heading-sans-serif"
            >
              Contact 2
            </div>
            <div
              class="pf-u-icon-color-light"
            >
              Intro 2
               ...
            </div>
          </div>
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ContactList renders with full contacts list 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="pf-c-list pf-m-plain"
      >
        <li
          class=""
        >
          <div
            class="pf-c-empty-state pf-m-xs"
          >
            <div
              class="pf-c-empty-state__content"
            >
              <div
                class="pf-c-empty-state__body"
              >
                No contacts found
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="pf-c-list pf-m-plain"
    >
      <li
        class=""
      >
        <div
          class="pf-c-empty-state pf-m-xs"
        >
          <div
            class="pf-c-empty-state__content"
          >
            <div
              class="pf-c-empty-state__body"
            >
              No contacts found
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
